### PR TITLE
HW8

### PR DESCRIPTION
**Pull request by Amudah41**

### CODE REVIEW

File: hw8/tasks/task82.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        cursor = self.conn.cursor()
+        cursor.execute(
+            f"SELECT * FROM {self.table_name} where name=:name", {"name": item_name}
+        )
+        return cursor.fetchone()
+
+    def __iter__(self) -> "TableData":
+        self.number_of_row = 0
+        self.colomns = self.name_of_table_colomns()
+        print(type(self))
```
Comment (AlekseiAfanasev) [inline hw8/tasks/task82.py]: forgot to remove probably

File: hw8/tasks/task82.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        return self
+
+    def name_of_table_colomns(self) -> List[str]:
+        cursor = self.conn.cursor()
+        cursor.execute(
+            f"SELECT name FROM PRAGMA_TABLE_INFO(:table)", {"table": self.table_name}
+        )
+        return [colomn[0] for colomn in cursor.fetchall()]
```
Comment (AlekseiAfanasev) [inline hw8/tasks/task82.py]: "Avoid reading entire table into memory." - from the task

File: hw8/tasks/task82.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    def __getitem__(self, item_name: str) -> Tuple[Any]:
+        cursor = self.conn.cursor()
+        cursor.execute(
+            f"SELECT * FROM {self.table_name} where name=:name", {"name": item_name}
+        )
+        return cursor.fetchone()
+
+    def __iter__(self) -> "TableData":
+        self.number_of_row = 0
+        self.colomns = self.name_of_table_colomns()
```
Comment (AlekseiAfanasev) [inline hw8/tasks/task82.py]: iter() function can help you here

File: hw8/tasks/task82.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+        return self
+
+    def name_of_table_colomns(self) -> List[str]:
+        cursor = self.conn.cursor()
+        cursor.execute(
+            f"SELECT name FROM PRAGMA_TABLE_INFO(:table)", {"table": self.table_name}
+        )
+        return [colomn[0] for colomn in cursor.fetchall()]
+
+    def __next__(self) -> Dict[str, Any]:
```
Comment (AlekseiAfanasev) [inline hw8/tasks/task82.py]: it is not needed here

Comment (Amudah41) [reply hw8/tasks/task82.py]: Done

File: hw8/tasks/task82.py (hunk 2 above)
Comment (Amudah41) [reply hw8/tasks/task82.py]: Fixed it

File: hw8/tasks/task82.py (hunk 3 above)
Comment (Amudah41) [reply hw8/tasks/task82.py]: Done

**Pull request merged**